### PR TITLE
Release: set version in source for releases

### PR DESCRIPTION
**Pull request by timmywil**
Fixes gh-2979

### CODE REVIEW

**Comment by mgol** [comment]
Could you also replace [`@DATE`](https://github.com/jquery/jquery/blob/59ec78e6020cc963c1f95cb96a28eaaf20e37b3e/src/wrapper.js#L13)?

File: build/release.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+var fs = require( "fs" );
```
Comment (markelog) [inline build/release.js]: Looks inconsistent, other requires in the `module.exports` scope

File: build/release.js [hunk 2]
```diff
@@ -27,6 +29,15 @@ module.exports = function( Release ) {
 			ensureSizzle( Release, callback );
 		},
 		/**
+		 * Set the version in the src folder for distributing AMD
+		 */
+		_setSrcVersion: function() {
+			var corePath = __dirname + "/../src/core.js",
+				contents = fs.readFileSync( corePath, "utf8" );
+			contents = contents.replace( /@VERSION/g, Release.newVersion );
```
Comment (markelog) [inline build/release.js]: Feels so weird that we use those regexes all over, while not using something builder, package or something

File: build/release.js (hunk 1 above)
Comment (timmywil) [inline build/release.js]: I make a distinction between modules that should be required when the exported function is called and ones that don't need to be.

File: build/release.js (hunk 2 above)
Comment (timmywil) [inline build/release.js]: I ain't afraid of regex replaces. :)

**Comment by timmywil** [comment]
@mgol I think it's okay to publish files that aren't used. If anyone tries to use wrapper.js, they'll find out quick it doesn't work, but I doubt anyone will actually try that. I don't mind leaving it out, but not copying it would mean switching to a list of all of the files in the src folder, rather than just copying the src folder (at least for bower), and I'd rather not do that.

**Comment by mgol** [comment]
> I don't mind leaving it out, but not copying it would mean switching to a list of all of the files in the src folder, rather than just copying the src folder (at least for bower), and I'd rather not do that.

You could use glob patterns & negate this one file out without major changes. But I don't insist. ;)

So, are you for leaving it inside the published package with the non-replaced `@DATE`?

File: build/release.js (hunk 2 above)
Comment (markelog) [inline build/release.js]: <xkcd commic>

File: build/release.js (hunk 1 above)
Comment (markelog) [inline build/release.js]: Feels very specific as "gotcha" style rule i.e. hard to follow, don't recall we do this in anywhere else.

File: build/release.js (hunk 2 above)
Comment (dmethvin) [inline build/release.js]: :smile_cat: True but it's a question of complexity. I know how a regex works very well, but if you pull in a template package I will have to look up how it works. At some point I agree it might make sense to put one in, but to me it doesn't look like we are at that point yet. The code is all very understandable and we haven't had problems with it.

Comment (markelog) [inline build/release.js]: Somewhere i suggested to use underscore templates since they used in grunt itself and lodash (obviously) also has them, which make them kind of standard in some sense.

The bigger question for me is that we have a lot of custom logic for the build, i don't know, but it seems this should be a common issue and therefore should have a common solution...

But of course i need to put my money where the mouth is, i.e. do something about this, otherwise it is just unproductive critic from my side.

**Comment by timmywil** [comment]
> So, are you for leaving it inside the published package with the non-replaced @DATE?

For now, I'd be open to a later patch excluding the file, but it's not high priority.
